### ISSUE TITLE
Importing testtoolkit ends with timeout on schema synchronization

### ISSUE THREAD

**Issue by kine** [Microsoft/navcontainerhelper#163]
On some images I am getting this error when creating container including test toolkit>
```
: [31719426] The objects from the .fob file were imported, but they contain schema changes that cannot be synchronized to the database with the Microsoft Dynamics NAV Server instance.
Contact your system administrator with the following information:

Server Name:         localhost
Server Instance:     NAV
Tenant Name:

Description:
The requested task failed to start within the timeout period of 10 seconds that has been specified in the Microsoft Dynamics NAV Development Environment / Database / Alter. The
reason is that Microsoft Dynamics NAV Server is busy with a similar task.
 -- File: C:\TestToolKit\CALTestRunner.fob
```
I cannot see any way how to change the timeout. Right now I am getting this on image microsoft/dynamics-nav:2018-ch. It is mostly on images with heavy localization (more objects?).

**Comment by freddydk** [comment]
Does this also happen if you try to import this using CSIDE?


/Freddy

Fra: Kamil Sacek <notifications@github.com>
Sendt: 9. juli 2018 09:11
Til: Microsoft/navcontainerhelper <navcontainerhelper@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Emne: [Microsoft/navcontainerhelper] Importing testtoolkit ends with timeout on schema synchronization (#163)


On some images I am getting this error when creating container including test toolkit>

: [31719426] The objects from the .fob file were imported, but they contain schema changes that cannot be synchronized to the database with the Microsoft Dynamics NAV Server instance.

Contact your system administrator with the following information:



Server Name:         localhost

Server Instance:     NAV

Tenant Name:



Description:

The requested task failed to start within the timeout period of 10 seconds that has been specified in the Microsoft Dynamics NAV Development Environment / Database / Alter. The

reason is that Microsoft Dynamics NAV Server is busy with a similar task.

 -- File: C:\TestToolKit\CALTestRunner.fob

I cannot see any way how to change the timeout. Right now I am getting this on image microsoft/dynamics-nav:2018-ch. It is mostly on images with heavy localization (more objects?).

**Comment by MikeGlue** [comment]
I can confirm that I've seen this happen as well with the US images (NAV 2018 and BC are only ones I've been using, but all have been US). It's been hit and miss though as to when it happens. I've had it happen but then a re-run of my entire "create container" script would be ok. Even if the script fails though doing the import via C/Side has always worked.

**Comment by MikeGlue** [comment]
This is the other error that I periodically get when importing the test toolkit:
![2018-07-27 10_41_31-microsoft edge](https://user-images.githubusercontent.com/5204734/43327584-cbdecd50-9189-11e8-85c2-608e3e18bfeb.png)

**Comment by MikeGlue** [comment]
I used a modified command that did not generate the symbols and delayed the schema sync until all of the objects were imported, and now I can run my build and it passes this step every time! Code is below.....I can submit as a pull request if you want to make this change. The only downside is that the symbols will not generate, but I am not sure this is bad since the 'Test' symbols package is already loaded into the base database.

```
Get-ChildItem -Path "C:\TestToolKit\*.fob" | ForEach-Object { 
        $objectsFile = $_.FullName
        Write-Host "Importing Objects from $objectsFile (container path)"
        $databaseServerParameter = $databaseServer

        Import-NAVApplicationObject -Path $objectsFile `
                                    -DatabaseName $databaseName `
                                    -DatabaseServer $databaseServerParameter `
                                    -ImportAction Overwrite `
                                    -SynchronizeSchemaChanges No `
                                    -NavServerName localhost `
                                    -NavServerInstance NAV `
                                    -NavServerManagementPort 7045 `
                                    -Confirm:$false
        }
        
        #Sync after all objects hav been imported
        Sync-NavTenant NAV -Mode ForceSync -Force
```

**Comment by richardrobberse** [comment]
That's exactly what I did Mike, it works a lot better!

**Comment by hemisphera** [comment]
What is the status on this? I still have the same problem, yet I see that the pullrequest has been closed.

**Comment by freddydk** [comment]
A fix will be included in the next drop of navcontainerhelper (probably this weekend)

**Comment by freddydk** [comment]
Included in version 0.3.1.0
